src/mutual_inclusion_prototype/ReconstructedParticleStruct.jl

```diff
@@ -0,0 +1,22 @@
+using StaticArrays
+include("Vector3fConstructor.jl")
+include("ParticleIDConstructor.jl")
+include("ClusterConstructor.jl")
+include("TrackConstructor.jl")
+
+mutable struct ReconstructedParticle{T}
```

Comment:
```suggestion
mutable struct ReconstructedParticle{ClusterT, TrackT, ReconstructedParticleT, ParticleIDT, VertexT}
```
If we follow this all the way through for all relations this would look something like this, right? Obviously the constructor would then have to be
```julia
function ReconstructedParticle()
  return ReconstructedParticle{Cluster, Track, ReconstructedParticle, ParticleID, Vertex} (
    # ... populate ...
    )
end
```
The user would never have to see this, but it would make code generation a bit easier I could imagine.

Reply:
Yes, that make sense.